docs/project-file-structure.md

```diff
@@ -64,30 +70,136 @@ to organize their structure
 
 Resources, scenes, scripts etc. used exclusively within that scene should be stored in the scenes subdirectory
 
+A scene and its script should always be together within the directory i.e. no /scripts/ and /scenes/ directories should exist within
+the project somewhere.
+
 Example:
 ```
 root/
 |- src/
-    |- Theme/
+    |- theme/
     |   |- ButtonNormal.tres
     |   |- ButtonHover.tres
     |   |- ButtonDisabled.tres
-    |- Scripts/
+    |- utility/
     |   |- Math.gd
-    |- Menu/
+    |- menu/
     |   |- MenuManager.tscn
     |   |- MenuManager.gd
     |   |- MainMenu/
     |   |   |- MainMenu.tscn
     |   |   |- MainMenu.gd
     |   |   |- MainMenuFont.ttf
     |   |   |- MainMenuLabel.tres
-    |   |- SettingsMenu/
+    |   |- settings_menu/
     |   |   |- Settings.tscn
     |   |   |- Settings.gd
     |   |   |- Resolution/
     |   |   |   |- ResolutionSelector.gd
-    |   |   |- Volume/
+    |   |   |- volume/
     |   |   |   |- VolumeGrid.tscn
     |   |   |   |- VolumeGrid.gd
 ```
+
+#### Naming Convention
+
+Directory names should be snake_case.
+```
+root/src/game_session/
+```
+
+File names should be PascalCase.
+```
+root/src/game_session/Map.tscn
+```
+
+### Asset Organization
+
+Asset files should be organized within a relevant category from root/
+```
+root/art/
+root/music/
+root/sfx/
+```
```

Comment:
So similar to how I've said resources should be handled? Within their specific directory?

The main reason I'd argue against this is because of modding. We want people to be able to swap out assets ad nauseam. So therefore having those assets grouped in a specific directory means they can build distinct files that can be swapped in and out in essence. Also the vast majority of assets will get reused so it doesn't make sense to me to put them in the directory its gets used in, in the one or two case's it doesn't get reused.

Reply:
if the objective were to make them moddable outside of the pck file, we foremost need to declare a separation between root and data directories.

Reply:
OK so then user://, res:// and data://. Can we actually enforce that in godot or do we have to do it ourselves

Reply:
Considering this is the same convo as your other suggestion now, I'll mark this as resolved

Reply:
There is no `data://` protocol concept in Godot 4 rn, hopefully Godot 4.1 will have that.